Feature/data analysis and time controller

germanohn opened the pull request:
# Major Changes 

- Change library of charts from Vega to Recharts. Then, it removes all files that have specs. Also it removes files regarding layout, because this is not implemented yet with the new configuration.

- Add a new group of controllers called TimeControllers

- Split controllers into two big groups: 
    - EnvControllers: set the app and info concerning the app such as version, region and context
    - TimeControllers: set the time period in which the chart will be analyzed. It will have some options: DateRange, relative and presets. For now it has DateRange working.

- Add component DataAnalysis which is responsible to show charts from a app (layout). For now 
it has MemoryUsageLineChart done and tested.

- Set communication between Controllers and DataAnalysis through react contexts

Commit:
adicionando dropdown workspaces, ainda não funcionando

Commit:
workspaces dropdown

Refactor WorkspacePicker and AppsPicker to use a common function dropdownOptions
in utils

Commit:
Do tests with dropdowns in react/components/ControllersWrapper.tsx

Commit:
Create TimeController to set DateRange and DataAnalysis to create charts.

Main Changes:

- reorganize folders of components. Now is following:
    Metrics
        Controller
            EnvControllers
            TimeController
        DataAnalysis

- In EnvControllers, there are the controllers of
environment such as appPicker, version and region.
I just reorganize them.

- create TimeController with initially the layout of
a DateRange picker.

- create DataAnalysis that organizes and creates
charts. For now, there are Line and Bar chart with
some data from layoutContent (present in folder common)

- manifest.json:
remove builder of vega

- package.json:
add recharts

- global.d.ts
add vtex.styleguide and its components

- remove some unused files such as react/metrics

germanohn committed:
Link TimeControllers info with DataAnalysis through TimeContext

Changes:

- add TimeContext
react context that allows info from TimeControllers to be used. Now, the
changes the user makes in TimeControllers directly impacts the date range
of storedash request.

- change TimeControllers to use TimeContext
create a local state with provisory info (startDate, endDate and RangeStep)
of TimeContext that is manipulated by some components (RangeExtremesPicker
for example) and then create one button that pass these provisory info to
TimeContext

- use info from TimeContext in DataAnalysis
if there are values from TimeControllers (chosen by user), it changes
metricParams from storedash request

- remove unused Code
some code in Metrics/index.tsx that used spec from Vega.

germanohn committed:
Create chart for MemoryUsage metric

germanohn committed:
Update MemoryLineChart

Detailed changes:

- fix overflow in yAxis

- use abbrNum to show in a more friendly way numbers in xAxis

- add colors to lines

- fix xAxis to show hours

- made customTooltip

germanohn committed:
more changes in MemoryUsageLineChart

changes:
- add formatted time, which means that XAxis shows the unit metric
in interval (one metricParam)

- instead of showing a absolute value for memory usage, now it shows
the mean value

Besides, I change all object of type moment from moment.js to the native
Date from Javascript

germanohn committed:
Activate filters of envControllers for charts

Other changes:

- EnvControllers:
now envControllers use envContext to get and change state from Metrics,
instead of doing this by props.

- TimeControllers:
move up one layer the apply button and the provisory state, because dateRange,
relative and the future options of date period share these funcionalities

- Fix typings of Time and Env context

- Latency chart in progress

- change .js files to .tsx and set the appropriate types

germanohn committed:
Reorganize files, fix lint and ts errors and add typings

tlgimenes commented on package.json:
```diff
@@ -0,0 +1,5 @@
+{
```
please, remove this file

tlgimenes commented on react/common/layoutContent.ts:
```diff
@@ -0,0 +1,21 @@
+    'ChartType': MemoryUsageLineChart,
+    'storedash': {
+      'metricParams': {
+        'appName': 'render-server',
```
why not initializing with an empty string ?

tlgimenes commented on react/common/layoutContent.ts:
```diff
@@ -0,0 +1,21 @@
+
+export default [
+  {
+    'ChartType': MemoryUsageLineChart,
```
why not removing the `'` from the keys ?

tlgimenes commented on react/package-lock.json:
```diff
@@ -0,0 +1,1570 @@
+{
```
why do we need this file ?

tlgimenes commented on react/yarn-error.log:
```diff
@@ -0,0 +1,1067 @@
+Arguments: 
```
why versioning this file ?

tlgimenes commented on react/typings/global.d.ts:
```diff
@@ -21,28 +21,48 @@ interface RenderContext {
+
+type SetEnvControllers = (envControllers: EnvController) => void
+
+declare module 'vtex.styleguide' {
```
why not moving this to another file ?

tlgimenes commented on react/locales/en-US.json:
```diff
@@ -6,6 +6,7 @@
   "console.submenu.publishes": "Publishes",
   "console.admin.metrics.datePicker.startDateTitle": "Start date",
   "console.admin.metrics.datePicker.endDateTitle": "End date",
+  "console.admin.components.metrics.controllers": "Controllers",
```
why not using the new messages builder ?

tlgimenes commented on manifest.json:
```diff
@@ -9,14 +9,13 @@
   "builders": {
-    "vega": "0.x",
     "react": "2.x",
     "pages": "0.x"
```
why not using the new messages builder ?

tlgimenes commented on react/components/Metrics/Controllers/index.tsx:
```diff
@@ -0,0 +1,33 @@
+        variation="half"
+        titleAside={
+          <div className="nv4 flex justify-end">
+            <Button variation="primary" >
```
why leaving a button that has no effect ?

tlgimenes commented on react/components/Metrics/Controllers/EnvControllers/appPicker.tsx:
```diff
@@ -0,0 +1,30 @@
+import { EnvContext } from '../../Contexts/EnvContext'
+
+
+export default class AppPicker extends Component {
```
why not using `React.SFC` ?

tlgimenes commented on react/components/Metrics/Controllers/EnvControllers/appPicker.tsx:
```diff
@@ -0,0 +1,30 @@
+    <EnvContext.Consumer>
+      {({ envControllers, setEnvControllers }) => (
+        <Query query={appsQuery} ssr={false}>
+          {({ loading, data: { appsWithStats } }) => !loading && appsWithStats &&
```
why not using a preview when the query is loading ?

tlgimenes commented on react/components/Metrics/Controllers/EnvControllers/contextPicker.tsx:
```diff
@@ -0,0 +1,38 @@
+  }
+]
+
+export default class ContextPicker extends Component {
```
maybe environment picker ?

tlgimenes commented on react/components/Metrics/Controllers/EnvControllers/contextPicker.tsx:
```diff
@@ -0,0 +1,38 @@
+  {
+    label: 'Development',
+    value: false,
+  }
```
missing coma?

tlgimenes commented on react/components/Metrics/Controllers/EnvControllers/versionInput.tsx:
```diff
@@ -0,0 +1,73 @@
+  private parseInputAndSetState = async (version: string, envControllers: EnvController, setEnvControllers: SetEnvControllers) => {
+    const isValid = version.length === 0 || isValidVersion(version)
+    if (!isValid) {
+      await this.setState({ inputIsNotSemver: true })
```
setState returns a promise ?

germanohn replied:
it returns undefined. So I think it makes more sense to add a wrapper as it done here:

https://medium.com/front-end-weekly/async-await-with-react-lifecycle-methods-802e7760d802

The other option would be to put a callback in the second argument of setState as it is described in docs. I am also thinking in how to use with componentDidUpdate.

tlgimenes commented on react/components/Metrics/Controllers/TimeControllers/Relative/index.tsx:
```diff
@@ -0,0 +1,11 @@
+  public render = () => {
+    return (
+      <div className="pa4 mh2" >
+        placeholder for Relative mode
```
we can't release this to production with placeholders in the title of features. It looks like an unfinished work

tlgimenes commented on react/components/Metrics/Controllers/TimeControllers/DateRange/RangeStepPicker.tsx:
```diff
@@ -0,0 +1,80 @@
+  }
+
+  public render = () => {
+    const stepModifierOptions: string[] = [
```
why creating this array at every render ?

tlgimenes commented on react/components/Metrics/Controllers/TimeControllers/DateRange/RangeStepPicker.tsx:
```diff
@@ -0,0 +1,80 @@
+    if (this.state.stepModifier === 'Full') {
+      this.props.handleRangeStep('')
+    } else {
+      const mapTime = {
```
why creating this map at every function call ?

tlgimenes commented on react/components/Metrics/Controllers/TimeControllers/DateRange/typings.d.ts:
```diff
@@ -0,0 +1 @@
+export type RangeStep = 'Full' | 'Seconds' | 'Minutes' | 'Hours' | 'Days' | 'Months' | 'Years'
```
why using a .d.ts for this ?

tlgimenes commented on react/components/Metrics/DataAnalysis/index.tsx:
```diff
@@ -0,0 +1,80 @@
+                        },
+                      } = chartDescription
+
+                      metricParams = this.updateEnvironment(metricParams, envControllers)
```
this name makes me think in another thing

tlgimenes commented on react/components/Metrics/DataAnalysis/CustomElements/CustomTooltip.tsx:
```diff
@@ -0,0 +1,85 @@
+            </div>
+            <div className="flex justify-between" >
+              <div className="pa2" style={{ color: payload[0].color }} >
+                {`${this.handleLabel(payload[0].dataKey)}:  `}
```
why not using `<FormattedMessage>` directly

germanohn replied:
Actually, I think it is better to maintain in english to info concerning the metric. For example, MemoryUsage has info such as: external, rss, heapUsed and heapTotal. Some of them does not make sense if translated.

But I'll change to use <FormattedNumber> instead of handleValue

germanohn committed:
Add templates for Issues and PR

germanohn committed:
Remove package.json

germanohn committed:
Remove info that should not be in metricParams

germanohn committed:
Add messages and extract dataAnalysis from utils

germanohn committed:
Turn stateless class components into stateless functional components

germanohn committed:
Add type RangeStep to RangeStepPicker

germanohn committed:
Put mapTime and stepModifierOptions as global variables

germanohn committed:
Add wrapper to setState

germanohn committed:
Disabled unworking new metric button

germanohn committed:
Change functions signatures

germanohn committed:
Change EmptyStates messages for comming features

germanohn committed:
Change handleValue by FormattedNumber

germanohn committed:
Move vtex.styleguide module to a separate file

germanohn committed:
Update gitignore to ignore .log and .lock files

germanohn committed:
change gitignore and delete react/yarn-error.log

germanohn committed:
Rename contextPicker for EnvPicker and others env for app

germanohn committed:
Put internationalization in AppControllers

alcar commented on react/Console.tsx:
```diff
@@ -8,15 +8,15 @@ import Errors from './components/Errors'
+  & {
+    params: {
+      tab: string
+    }
```
:pick: This could be written as:
```ts
interface Props extends ReactIntl.InjectedIntlProps {
  params: {
    tab: string
  }
  runtime: RenderContext
}
```

germanohn replied:
ok

alcar commented on react/Console.tsx:
```diff
@@ -34,44 +34,44 @@ const fields: Field[] = [
 class Console extends Component<Props, {}> {
   constructor(props: any) {
     super(props)
   }
```
:pick: Since nothing is being done after calling `super(props)`, there's no need to have an explicit constructor definition.

alcar commented on react/Console.tsx:
```diff
@@ -34,44 +34,44 @@ const fields: Field[] = [
+                key={name}
+                label={intl.formatMessage({ id: titleId })}
+                active={tab === path}
+                onClick={() => navigate({ to: `/admin/console/${path}` })}
```
Since you're using a class component, I'd move this click handler to a method in order to prevent the creation of a new function on each render.

germanohn replied:
How to do that, since i have to pass navigate and path as arguments

alcar commented on react/common/constants.ts:
```diff
@@ -0,0 +1,4 @@
+export const CHART_PROPERTIES = {
+  height: 400,
```
:pick: I would explicitly write '400px'.

thiagomurakami replied:
Since this is a React default, I think this is fine.

germanohn replied:
for clarity I think it makes sense

germanohn replied:
Actually, according to recharts API, height and width are Percentage | Number. 

http://recharts.org/en-US/api/ResponsiveContainer#width

alcar commented on react/common/dataAnalysis.ts:
```diff
@@ -0,0 +1,103 @@
+
+export const isNilOrEmpty = (x: any) => isEmpty(x) || isNil(x)
+
+const getFormattedTime = (date: any, intl: any, stepModifier: string) => {
```
:pick: The `intl` type is  `InjectedIntl` from `react-intl`.

alcar commented on react/common/dataAnalysis.ts:
```diff
@@ -0,0 +1,103 @@
+  }
+}
+
+export const addFormattedTime = (chartData: any, intl: any, stepModifier: string) => {
```
:pick: The `intl` type is `InjectedIntl` from `react-intl`.

alcar commented on react/common/utils.ts:
```diff
@@ -0,0 +1,24 @@
+  uniqWith(strEq)
+)(options)
+
+export const formattedDropdownOptions = (options: Option[], intl: any) => map(
```
:pick: The `intl` type is `InjectedIntl` from `react-intl`.

alcar commented on react/components/Errors.tsx:
```diff
@@ -1,12 +1,15 @@
     <EmptyState title="Coming Soon">
       <p>
-        Formula VTEX should last LONGER !!!.
+        Comming soon!
```
`Comming` => `Coming`. Also, this is not internationalized.

germanohn replied:
ok

alcar commented on react/components/Metrics/Contexts/AppContext.tsx:
```diff
@@ -0,0 +1,20 @@
+    production: 'true',
+    region: 'Any',
+  },
+  setAppControllers: (appControllers: AppController) => undefined,
```
Is the return type `undefined`? Perhaps you meant:
```ts
setAppControllers: (appControllers: AppController) => void
```

alcar commented on react/components/Metrics/Contexts/TimeContext.tsx:
```diff
@@ -0,0 +1,18 @@
+
+
+export const TimeContext = React.createContext<TimeContextInterface>({
+  setTimeControllers: (timeControllers: TimeController) => undefined,
```
Is the return type `undefined`? Perhaps you meant:
```ts
setTimeControllers: (timeControllers: TimeController) => void
```

germanohn replied:
The type of setTimeControllers is defined by TimeContextInterface, this

```ts
(timeControllers: TimeController) => undefined
```

just defines the value of setTimeControllers

alcar replied:
But is it actually a function that returns `undefined`? If there are no `return` statements, the return type should be `void`.

alcar commented on react/components/Metrics/Controllers/AppControllers/VersionInput.tsx:
```diff
@@ -0,0 +1,81 @@
+    if (!isValid) {
+      await this.setStateAsync({ inputIsNotSemver: true })
+    } else {
+      await this.setStateAsync({ inputIsNotSemver: false })
```
I don't think you should be doing this. React's `setState` [may batch or defer the update until later](https://reactjs.org/docs/react-component.html#setstate), so awaiting for a `setState` could lead to unwanted behaviour. What you could do is use its second argument, the callback, to do something after the new state is set. However, in this case, I believe you don't need to keep the validity of the input in the state, since it can be easily determined. I would move the `isValid` check to the `render` method, and send it as an argument to `parseInput`.

germanohn replied:
I agree. I've tried your suggestion doing the following:

```ts
let inputIsSemver: boolean = true
return (
    <AppContext.Consumer>
        {({ appControllers, setAppControllers }) => (
        <Input
            placeholder="Ex: 1.0.x"
            label={this.props.intl.formatMessage({ id: 'console.app.version' })}
            onChange={(event: any) => {
            inputIsSemver = this.parseInput(event.target.value, appControllers, setAppControllers)
            
            }}
            errorMessage={inputIsSemver ? ''" : 'Version is not in the SemVer spec'}
        />
        )}
    </AppContext.Consumer>
)
```

but despite inputIsSemver is changing its value according to this.parseInput return, the errorMessage does not change. So, I decide to remove async/await and use the second argument callback of setState

alcar replied:
My suggestion was actually to move

```ts
const isValid = version.length === 0 || isValidVersion(version)
```

to the `render` method, but I just realized that `version` comes from the `onChange` `event`. Sorry.

In that case, I think your callback approach sounds good!

alcar commented on react/components/Metrics/Controllers/TimeControllers/DateRange/RangeStepPicker.tsx:
```diff
@@ -0,0 +1,83 @@
+    }
+  }
+
+  public constructRangeStep = () => {
```
This method doesn't need to be public.

germanohn replied:
Ok

alcar commented on react/components/Metrics/Controllers/TimeControllers/DateRange/RangeStepPicker.tsx:
```diff
@@ -0,0 +1,83 @@
+        <NumericStepper
+          label="Step"
+          size="small"
+          onChange={(e: EventWithValue) => this.setState({ stepSize: e.value })}
```
Since you're using a class component, I'd move this change handler to a method in order to prevent the creation of a new function on each render.

germanohn replied:
I investigate the code from styleguide, but despite managing to change for RadioGroup for example, for this one I didn't find a workaround, so I decide to keep this way.

alcar commented on react/components/Metrics/Controllers/TimeControllers/DateRange/RangeStepPicker.tsx:
```diff
@@ -0,0 +1,83 @@
+          size="small"
+          options={dropdownOptions(stepModifierOptions)}
+          value={this.state.stepModifier}
+          onChange={(e: Event, stepModifier: RangeStep) => this.setState({ stepModifier })}
```
Since you're using a class component, I'd move this change handler to a method in order to prevent the creation of a new function on each render.

alcar commented on react/components/Metrics/Controllers/TimeControllers/Relative/index.tsx:
```diff
@@ -0,0 +1,12 @@
+const Relative = () => {
+  return (
+    <div className="pa4 mh2" >
+      Comming soon!
```
`Comming` => `Coming`. Also, this is not internationalized.

germanohn replied:
Ok

alcar commented on react/components/Metrics/Controllers/TimeControllers/index.tsx:
```diff
@@ -0,0 +1,146 @@
+              <Fragment>
+                <EmptyState title="Please set a time range">
+                  <p>
+                    Please set a time range to analyze your metric
```
This is not internationalized.

alcar commented on react/components/Metrics/Controllers/TimeControllers/index.tsx:
```diff
@@ -0,0 +1,146 @@
+                    size="small"
+                    onClick={() => this.setTimeMode('DateRange')}
+                  >
+                    Set Time Range
```
This is not internationalized.

alcar commented on react/components/Metrics/Controllers/TimeControllers/index.tsx:
```diff
@@ -0,0 +1,146 @@
+    }
+  }
+
+  public setStartDate = (startDate: Date) => {
```
This method doesn't need to be public.

alcar commented on react/components/Metrics/Controllers/TimeControllers/index.tsx:
```diff
@@ -0,0 +1,146 @@
+    this.setState({ startDate })
+  }
+
+  public setEndDate = (endDate: Date) => {
```
This method doesn't need to be public.

alcar commented on react/components/Metrics/Controllers/TimeControllers/index.tsx:
```diff
@@ -0,0 +1,146 @@
+    this.setState({ endDate })
+  }
+
+  public setRangeStep = (rangeStep: string) => {
```
This method doesn't need to be public.

alcar commented on react/components/Metrics/Controllers/TimeControllers/index.tsx:
```diff
@@ -0,0 +1,146 @@
+    this.setState({ rangeStep })
+  }
+
+  public isButtonActive = () => {
```
This method doesn't need to be public.

alcar commented on react/components/Metrics/Controllers/TimeControllers/index.tsx:
```diff
@@ -0,0 +1,146 @@
+      endDate <= currentTime)
+  }
+
+  public setTimeMode = (mode: any) => {
```
This method doesn't need to be public.

alcar commented on react/components/Metrics/Controllers/TimeControllers/index.tsx:
```diff
@@ -0,0 +1,146 @@
+                  <Button
+                    variation="primary"
+                    size="small"
+                    onClick={() => this.setTimeMode('DateRange')}
```
Since you're using a class component, I'd move this click handler to a method in order to prevent the creation of a new function on each render.

alcar commented on react/components/Metrics/Controllers/TimeControllers/index.tsx:
```diff
@@ -0,0 +1,146 @@
+                      { value: 'Relative', label: 'Relative' },
+                    ]}
+                    value={this.state.mode}
+                    onChange={(e: EventWithValue) => this.setState({
```
Since you're using a class component, I'd move this change handler to a method in order to prevent the creation of a new function on each render.

alcar commented on react/components/Metrics/Controllers/TimeControllers/index.tsx:
```diff
@@ -0,0 +1,146 @@
+                            variation="primary"
+                            size="small"
+                            disabled={!this.isButtonActive()}
+                            onClick={() => {
```
Since you're using a class component, I'd move this click handler to a method in order to prevent the creation of a new function on each render.

alcar commented on react/components/Metrics/DataAnalysis/Charts/CpuUsageLineChart.tsx:
```diff
@@ -0,0 +1,93 @@
+
+  return (
+    <Fragment>
+      <BlockTitle title="Cpu usage (microseconds) over Time" />
```
This is not internationalized.

alcar commented on react/components/Metrics/DataAnalysis/Charts/LatencyLineChart.tsx:
```diff
@@ -0,0 +1,87 @@
+
+  return (
+    <Fragment>
+      <BlockTitle title="Latency (miliseconds) over Time" />
```
This is not internationalized.

alcar commented on react/components/Metrics/DataAnalysis/Charts/LatencyLineChart.tsx:
```diff
@@ -0,0 +1,87 @@
+                    />
+                    <Legend />
+                    <Tooltip content={<CustomTooltip />} />
+                    {/* { this.drawLatencyLine() } */}
```
Commented-out code.

germanohn replied:
But this is not done yet. This component was in progress in this PR

alcar commented on react/components/Metrics/DataAnalysis/Charts/MemoryUsageLineChart.tsx:
```diff
@@ -0,0 +1,98 @@
+
+  return (
+    <Fragment>
+      <BlockTitle title="Memory (bytes) Consumption over Time" />
```
This is not internationalized.

alcar commented on react/components/Metrics/index.tsx:
```diff
@@ -0,0 +1,83 @@
+    }
+  }
+
+  public setAppControllers = (appControllers: AppController) => {
```
This method doesn't need to be public.

alcar commented on react/components/Metrics/index.tsx:
```diff
@@ -0,0 +1,83 @@
+
+
+class Metrics extends Component<{}, State> {
+  constructor(props: any) {
```
:pick: The type of `props` is `{}`.

alcar commented on react/components/Metrics/index.tsx:
```diff
@@ -0,0 +1,83 @@
+    this.setState({ appControllers })
+  }
+
+  public setTimeControllers = (timeControllers: TimeController) => {
```
This method doesn't need to be public.

alcar commented on react/components/Metrics/index.tsx:
```diff
@@ -0,0 +1,83 @@
+                ? (
+                  <DataAnalysis layout={layoutContent} />
+                ) : (
+                  <EmptyState title="Please select an app">
```
This is not internationalized.

alcar commented on react/components/Metrics/index.tsx:
```diff
@@ -0,0 +1,83 @@
+                ) : (
+                  <EmptyState title="Please select an app">
+                    <p>
+                      Please select an app to be analyzed
```
This is not internationalized.

thiagomurakami commented on react/Console.tsx:
```diff
@@ -34,44 +34,44 @@ const fields: Field[] = [
     window.postMessage({ action: { type: 'STOP_LOADING' } }, '*')
   }
 
   public render = () => {
```
Unnecessary arrow function

thiagomurakami commented on react/components/Metrics/DataAnalysis/CustomElements/BlockTitle.tsx:
```diff
@@ -0,0 +1,13 @@
+}
+
+
+const BlockTitle = ({ title }: Props) => (
```
You can use
```
const BlockTitle: React.SFC<Props> = ...
```

thiagomurakami commented on react/components/Metrics/DataAnalysis/CustomElements/CustomTooltip.tsx:
```diff
@@ -0,0 +1,68 @@
+  if (active) {
+    const { payload, label } = props
+
+    return (
```
You could declare
```
return active && (...)
```

thiagomurakami commented on react/components/Metrics/DataAnalysis/CustomElements/CustomXAxisTick.tsx:
```diff
@@ -0,0 +1,19 @@
+}
+
+
+const CustomXAxisTick = ({ x, y, payload }: Partial<Props>) => (
```
Why are you using `Partial` here?

(You can use `React.SFC` to declare the type here.)

thiagomurakami commented on react/components/Metrics/DataAnalysis/CustomElements/CustomXAxisTick.tsx:
```diff
@@ -0,0 +1,19 @@
+interface Props {
+  x: number
+  y: number
+  payload: any
```
👀

germanohn replied:
In the @types/recharts I didn't find a interface that have those props

thiagomurakami commented on react/components/Metrics/DataAnalysis/CustomElements/CustomTooltip.tsx:
```diff
@@ -0,0 +1,68 @@
+interface Props extends InjectedIntlProps {
+  active?: boolean
+  label?: string
+  payload?: any
```
👀

germanohn replied:
Ok

thiagomurakami commented on react/typings/events.d.ts:
```diff
@@ -0,0 +1,8 @@
+export interface EventWithValue extends Event {
```
Aren't there other `Events` you could have used? Maybe `React.SyntheticEvent`, `React.MouseEvent`, `React.PointerEvent`, etc...

germanohn replied:
Yes, now I'm using React.ChangeEvent for the event in component RadioGroup from styleguide. But for the component NumericStepper I have to maintain EventWithValue.

thiagomurakami reviewed (commented):
Why is the mean being calculated on the front? Shouldn't this come from the back-end?

germanohn committed:
Make suggested changes

Main changes:
- finish to internationalize messages
- change interface props in console
- Use the React.SFC syntax in functions with Props
- turn unnecessary public methods into private
- try to avoid arrow functions in onClick and onChange
- make NumericStepper block when stepModifier === 'Full'
- remove unnecessary event typings
- fix props in CustomTooltip and optimize the component using a map
- change render and lifecycles methods from arrow to normal functions

germanohn committed:
Move variables to utils and create noOp function for empty onChange of NumericStepper

germanohn committed:
Remove unnecessary functions in utils
